dev/Pager/PrototypePager/PrototypePagerElementFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace MUXControlsTestApp
+{
+    public class PrototypePagerElementFactory : ElementFactory
+    {
+        protected override UIElement GetElementCore(Microsoft.UI.Xaml.Controls.ElementFactoryGetArgs args)
```

Comment:
You should be able to achieve this is a much more elegant fashion by using DataTemplateSelector.

Reply:
But don't feel that you have to do that if you don't have the time. It is just a suggestion. The current solution will work also.

---
In reply to: [478751590](https://github.com/microsoft/microsoft-ui-xaml/pull/3196#discussion_r478751590) [](ancestors = 478751590)

Reply:
Cool! I'll defintely look into this later if I have time, but for now I will stick with the element factory!